app/controllers/opendata/dataset/resources_controller.rb

```diff
@@ -10,95 +10,95 @@ class Opendata::Dataset::ResourcesController < ApplicationController
-      raise "403" unless @dataset.allowed?(:read, @cur_user, site: @cur_site, node: @cur_node)
-      @item = @dataset.resources.find params[:resource_id]
-      send_file @item.file.path, type: @item.content_type, filename: @item.filename,
-        disposition: :attachment, x_sendfile: true
-    end
-
-    def download_tsv
-      @item = @dataset.resources.find params[:resource_id]
-      raise "404" if @item.tsv.blank?
-      raise "403" unless @dataset.allowed?(:read, @cur_user, site: @cur_site, node: @cur_node)
-      send_file @item.tsv.path, type: @item.content_type, filename: @item.tsv.filename,
-        disposition: :attachment, x_sendfile: true
-    end
-
-    def content
-      raise "403" unless @dataset.allowed?(:read, @cur_user, site: @cur_site, node: @cur_node)
-      @item = @dataset.resources.find params[:resource_id]
-      @data = @item.parse_tsv
-    end
-
-    def check_for_update
-      set_item
-      raise "403" unless @dataset.allowed?(:edit, @cur_user, site: @cur_site, node: @cur_node)
-
-      render layout: "ss/ajax"
-    end
-
-    def sync
-      set_item
-      raise "403" unless @dataset.allowed?(:edit, @cur_user, site: @cur_site, node: @cur_node)
-
-      return if request.get?
-
-      page = @item.assoc_page
-      file = @item.assoc_file
-      @item.update_resource_with_file!(page, file, @item.license_id)
-      render_update true, render: { file: :sync }
-    rescue => e
-      @item.errors.add :base, e.message
-      render_update false, render: { file: :sync }
-    end
+  def index
+    raise "403" unless @dataset.allowed?(:read, @cur_user, site: @cur_site, node: @cur_node)
+    @items = @dataset.resources.
+      search(params[:s]).
+      order_by(name: 1).
+      page(params[:page]).per(50)
+  end
+
+  def create
+    raise "403" unless @dataset.allowed?(:edit, @cur_user, site: @cur_site, node: @cur_node)
+
+    @item = @dataset.resources.new get_params
+    @item.status = params[:item][:state]
+    @item.workflow = { workflow_reset: true } if @dataset.member.present?
+    render_create @item.save
+  end
+
+  def update
+    raise "403" unless @dataset.allowed?(:edit, @cur_user, site: @cur_site, node: @cur_node)
+
+    @item.attributes = get_params
+    @item.status = params[:item][:state]
+    @item.workflow = { workflow_reset: true } if @dataset.member.present?
+    render_update @item.update
+  end
+
+  def download
+    raise "403" unless @dataset.allowed?(:read, @cur_user, site: @cur_site, node: @cur_node)
+    @item = @dataset.resources.find params[:resource_id]
+    send_file @item.file.path, type: @item.content_type, filename: @item.filename,
+      disposition: :attachment, x_sendfile: true
+  end
+
+  def download_tsv
+    @item = @dataset.resources.find params[:resource_id]
+    raise "404" if @item.tsv.blank?
+    raise "403" unless @dataset.allowed?(:read, @cur_user, site: @cur_site, node: @cur_node)
+    send_file @item.tsv.path, type: @item.content_type, filename: @item.tsv.filename,
```

Comment:
__[brakeman]__
* Parameter value used in file name [:ref](http://brakemanscanner.org/docs/warning_types/file_access/)

<!-- Comment by [SideCI](https://sideci.com) -->